tests/pytorch/test_numerics.py

```diff
@@ -1447,7 +1449,11 @@ def _test_grouped_linear_accuracy(block, num_gemms, bs, dtype, config, recipe, f
     outputs = [out, inp_hidden_states.grad]
     for p in block.parameters():
         if p.requires_grad:
-            outputs.append(p.grad)
+            if getattr(p, "main_grad", None) is not None:
+                outputs.append(p.main_grad)
+                assert p.grad is None  # grad should be None if fuse_wgrad_accumulation is True
```

Comment:
It turns out Mcore expects `p.grad` to not be `None`: https://github.com/NVIDIA/TransformerEngine/pull/1474#issuecomment-2652816973
https://github.com/NVIDIA/TransformerEngine/pull/1474 sets `grad` to an uninitialized tensor and assumes Mcore will ignore it.